pkg/apis/contrail/v1alpha1/config_types.go

```diff
@@ -687,7 +687,21 @@ func (c *Config) UpdateSTS(sts *appsv1.StatefulSet, instanceType string, request
 
 // SetInstanceActive sets the Cassandra instance to active
 func (c *Config) SetInstanceActive(client client.Client, activeStatus *bool, sts *appsv1.StatefulSet, request reconcile.Request) error {
-	return SetInstanceActive(client, activeStatus, sts, request, c)
+	if err := client.Get(context.TODO(), types.NamespacedName{Name: sts.Name, Namespace: request.Namespace},
+		sts); err != nil {
+		return err
+	}
+
+	*activeStatus = false
+	var acceptableReadyReplicaCnt = *sts.Spec.Replicas/2 + 1
```

Comment:
Pls, check if replicas pointer is not nil.

Reply:
Fixed